crates/typst/src/image/raster.rs

```diff
@@ -47,7 +58,9 @@ impl RasterImage {
         }
         .map_err(format_image_error)?;
 
-        Ok(Self(Arc::new(Repr { data, format, dynamic, icc })))
+        let dynamic = Arc::new(dynamic);
+        let encoded = encode_image(&dynamic, is_pdf, format);
```

Comment:
images are encoded once it is created, and we rely on this to get concurrency

Reply:
Yes, and the `OnceCell` will store the result once the computation is done. Now it's done directly in typst-pdf and it actually shows that we could do this for a lot more things 😈